lib/google_search_data_viewer_web/plugs/ensure_authenticated.ex

```diff
@@ -0,0 +1,27 @@
+# coveralls-ignore-start
+defmodule GoogleSearchDataViewerWeb.EnsureAuthenticatedPlug do
+  import Plug.Conn
+  import Phoenix.Controller
+
+  alias GoogleSearchDataViewerWeb.Router.Helpers, as: Routes
+
+  def init(opts), do: opts
+
+  def call(conn, _opts) do
+    user_id = get_session(conn, :user_id)
+
+    user = user_id && GoogleSearchDataViewer.Accounts.get_user(user_id)
```

Comment:
Utilizing pattern matching over here so we can use it with pipe operator and would make the code cleaner.
```elixir
def call(conn, _opts) do
  conn
  |> get_session(:user_id)
  |> get_user()
  |> case do
    nil -> # Handle unauthenticated case
    user -> conn
  end
end

defp get_user(nil), do: nil

defp get_user(user_id), do: GoogleSearchDataViewer.Accounts.get_user(user_id)
```